src/app/dynamic-forms/cinchy-dynamic-forms.component.ts

```diff
@@ -117,312 +140,300 @@ export class CinchyDynamicFormsComponent implements OnInit, OnChanges {
-    }
-  }
+            switch (field.cinchyColumn.dataType) {
+              case "Link":
 
+                if (field.cinchyColumn.isMultiple) {
+                  if (field.dropdownDataset?.options.length) {
+                    const valueSet: Array<any> = (Array.isArray(field.value)) ? field.value : [field.value];
 
-  handleOnFilter(filterText: string): void {
+                    const dropdownResult = field.dropdownDataset.options.filter((option: DropdownOption) => {
 
-    this.onLookupRecordFilter.emit(filterText);
-  }
+                      return valueSet.find((value: any) => {
 
+                        return (option.id === value);
+                      });
+                    });
 
-  setNewRow(newIndex: number): void {
+                    rowData[field.cinchyColumn.name] = dropdownResult.map((option: DropdownOption) => {
 
-    const newSelectedRow = this.lookupRecordsList[newIndex];
-    this.currentRow = newSelectedRow;
-    this.rowSelected(newSelectedRow);
-  }
+                      return option.id;
+                    }).join(",");
 
+                    rowData[`${field.cinchyColumn.name} label`] = dropdownResult.map((option: DropdownOption) => {
 
-  //#region Edit Add Child Form Data
-  async openChildForm(data) {
+                      return option.label;
+                    }).join(",");
+                  }
+                  else {
+                    rowData[field.cinchyColumn.name] = field.value ?? "";
+                  }
 
-    if (!this.isCloneForm && !this.rowId) {
-      const formvalidation = this.form.checkFormValidation();
-      if (formvalidation) {
-        this.saveForm(this.form, this.rowId, data);
-      }
-    } else {
-      this.openChildDialog(data);
-    }
-  }
+                  break;
+                }
+
+              // falls through
+              case "Choice":
+                const dropdownResult = field.dropdownDataset?.options?.find((option: DropdownOption) => {
+
+                  return (option.id === field.value);
+                });
 
+                if (dropdownResult) {
+                  rowData[field.cinchyColumn.name] = dropdownResult.id;
+                  rowData[`${field.cinchyColumn.name} label`] = dropdownResult.label;
+                }
+                else {
+                  rowData[field.cinchyColumn.name] = "";
+                }
 
-  afterChildFormEdit(eventData, childForm): void {
+                break;
+              case "Binary":
+                rowData[field.cinchyColumn.name] = field.value;
+                rowData[`${field.cinchyColumn.name}_Name`] = field.cinchyColumn.fileName;
 
-    this.childForms = childForm;
+                break;
+              case "Yes/No":
+                rowData[field.cinchyColumn.name] = field.value ?? false;
 
-    if (isNullOrUndefined(eventData)) {
-      return;
-    }
+                break;
+              default:
+                rowData[field.cinchyColumn.name] = field.value ?? "";
```

Comment:
Here as well, like mentioned in the other comment for Yes/No columns, do we want to preserve the null or treat nulls as empty strings like we're doing here?